src/core/rowindex_array.cc

```diff
@@ -314,40 +313,47 @@ void ArrayRowIndexImpl::compactify()
   _resize_data();
 }
 
+template<typename TI>
+void get_next_index_to_skip(const TI* inputs, size_t& j)
+{
+  while (inputs[j] == inputs[j+1]) {
+    ++j;
+  }
+  ++j;
+}
 
 template <typename TI, typename TO>
 RowIndexImpl* ArrayRowIndexImpl::negate_impl(size_t nrows) const
 {
   auto inputs = static_cast<const TI*>(buf_.rptr());
-  size_t newsize = nrows - length;
+  size_t newsize = nrows;
   size_t inpsize = length;
   Buffer outbuf = Buffer::mem(newsize * sizeof(TO));
   auto outputs = static_cast<TO*>(outbuf.xptr());
   TO orows = static_cast<TO>(nrows);
 
   TO next_index_to_skip = static_cast<TO>(inputs[0]);
-  size_t j = 1;  // next index to read from the `inputs` array
+  size_t count_indices_to_skip = 0;
```

Comment:
This seems more like `count_indices_skipped`

Reply:
Agree